compiler/rustc_session/src/session.rs

```diff
@@ -321,6 +325,23 @@ impl Session {
     ) -> DiagnosticBuilder<'_> {
         self.diagnostic().struct_span_warn_with_code(sp, msg, code)
     }
+    pub fn span_lint<S: Into<MultiSpan>>(&self, sp: S, lint: &'static Lint) {
```

Comment:
This shouldn't be necessary, see how `buffer_lint` is used in other built-in macros for reporting e.g. `BAD_ASM_STYLE`.

(Also, this is not a future-compatibility lint.)